### DISCUSSION TITLE
Welcome to Courseplay_FS22 Discussions!

### DISCUSSION THREAD

**Discussion by Tensuko** [Courseplay/Courseplay_FS22#540] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by XPiRiJenZ** [comment]
Hello :)
I'm using CP since its first release in FS11 and got addicted since then ;)
Also I'm a YT content creator (in german), who is creating easy understandable CP-tutorials since FS17 and now reporting about the development of CP in FS22 for each release version or interesting experimental.
In addition, there are letsplay videos for the FS, where the use of CP (and AD) is shown in practice.

If you're interested: www.youtube.com/c/xpirijenz

Cheers :)

**Reply by rohne83** [reply]
Hi,
I started playing FS end of summer 2020 (in fact FS was the first game I bought after nearly 20 years), and the availability of CP was one of the main reasons for me to start at all. Due to personal and family background I prefer the large scale farming practiced in North America, Australia and Eastern Germany - and as single player this would not work at all without CP ;)
Additionally I spend much time with modding (started to release those at www.ls-modcompany.com), with a focus on Eastern German vehicles (Fortschritt and IFA), but also working on newer vehicles. And always keeping an extra eye that those mods are working well with CP and AD.

**Comment by shadovwv** [comment]
Hi, I don't know if this is a good place to write here but I made autoupdater in exe (inside edited powershell script) file for Courseplay_FS22, maybe someone wanna have it for faster update, and many many thanks for this awsome mod You are the best. I don't play FS without it. It's like "must have it". PS. Sorry for my english it's not my native language. Greetings from Poland. Link for file http://modsfile.com/5gha8j9qhbs7/FS22_Courseplay_Updater.zip.html

**Reply by Lactic68** [reply]
Hi @shadovwv,
Thank you for your application.
I just have tested your CP Updater and everything seems to work fine but this program looks only on [user root folder]\Documents\My Games\FarmingSimulator2022\mods.
Is there a way to specify another folder to look at ?

**Reply by shadovwv** [reply]
Hi @Lactic68, yes You can change it at 4 line of powershell script. Link for script https://gist.github.com/shadovwv/abdaba0a5cbbd06ee5a3f936b367eb6a and look here #642

**Reply by Lactic68** [reply]
Nice thank you for your answer, so EXE is not needed anymore ?

**Reply by shadovwv** [reply]
I made an exe for easy use and had a nice look, to be not a just a regular script, and so as not to accidentally modify. You can use whatever You like exe or script.

**Comment by 112Farid** [comment]
Hallo,

ich ahbe ein Problem mit Cp. ICh kann keine Kurse generieren, ich aheb gar nicht die Möglichkeit einen Kurs (außer wegekurs) zu erstellen was man auch in dem bild sehen kann wasw im anhang ist kann mir einer bitte weiterhelfen.
lg
![Bild1](https://user-images.githubusercontent.com/88099374/159007324-b3c87269-2a3d-4a92-861a-53d52033a276.png)

**Reply by Tensuko** [reply]
Ich glaube hier fehlt Basiswissen. Tutorials auf YouTube von Mario Hirschfeld anschauen.
Hilfemenü kommt auch bald.

**Reply by 112Farid** [reply]
Ich habe mir das Video von MArio bereits angeschaut und alles genau so gemacht wie er ees gesagt hat. Ich habe keine Möglichkeit Feldarbeit einzustellen

**Reply by Tensuko** [reply]
> Ich habe mir das Video von MArio bereits angeschaut und alles genau so gemacht wie er ees gesagt hat. Ich habe keine Möglichkeit Feldarbeit einzustellen

Bezweifle ich stark.
Es funktioniert ja bei allen.
Vlt solltest du dich noch mal weiter damit beschäftigen...

**Comment by Wowfaktor** [comment]
Funktioniert der CP mod auf einem Dedi-Server ?
MvG.

**Reply by Tensuko** [reply]
Lies doch einfach mal etwas....
Oben auf Code klicken und runter scrollen...

**Reply by Wowfaktor** [reply]
Lieber Support, es hilft nicht sich den selben Text 10 mal durchzulesen, aber ich denke mal ihr wusstet dass schon, ich wollte euch nur mal drauf hinweien. 
Ich kann euren Support bedauerlicherweise nicht Loben, ihr seid ja nicht nur eine schlechte hilfe für mich, sodern auch noch für andere. 
Wenn man sich die Kommentare von @112Farid  durchliest, bezeichnet ihr andere ja auch als dumm, habt ihr schonmal drüber nachgedacht dass ihr selber nicht die schlausten seid ?  ;)

**Reply by Tensuko** [reply]
Als aller erstes, bezeichnen wir hier niemanden als dumm, eher als zu faul zum lesen.
Zweitens, braucht man einen Text nicht 10 mal durchlesen, wenn ganz oben in fetter Schrift etwas steht an der Stelle die ich dir beschrieben habe:
![MP Support](https://user-images.githubusercontent.com/27344744/159160152-3e1258b0-f20b-4310-ad03-46d1bda871bd.png)


Drittens, haben wir keine Glaskugel, wir kennen weder euren Wissensstand noch wo es hakte.

Uns für "nicht die schlauesten" zu halten ist kein cleverer Schachzug, wenn man von uns etwas wissen will.

Wir erwarten aber auch etwas eigeninitiative von den User, das heist sich selber mal schlau machen, selber was lesen, selber YouTube Videos anschauen und nicht bei jeder kleinigkeit Fragen stellen, die an andere Stellen schon 100 mal beantwortet wurden!

**Comment by Serg12-22** [comment]
The program is very good and necessary , I have been using it since FS 2013 , but I have a question why when installing a new version of CP , courses from the previous version disappear ?

**Reply by Tensuko** [reply]
Err, they don't ?
I test several Versions depending on the stuff we work on and I don't miss any courses.
They are all saved in a folder under modSettings\Courseplay or something.

(From our Wiki)
My Games\FarmingSimulator2022\modSettings\Courseplay_FS22\Courses\<map name>\Singleplayer

**Comment by Serg12-22** [comment]
I meant that they are not saved in the next version automatically , thanks for telling me where to look for them . 22.03.2022, 15:01, "Tensuko" ***@***.***>: Err, they don't ?I test several Versions depending on the stuff we work on and I don't miss any courses.They are all saved in a folder under modSettings\Courseplay or something. Would have to look it up.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

**Reply by Tensuko** [reply]
Still don't know what you mean.
No course is saved automatically.

**Comment by Serg12-22** [comment]
In FS 19 , when changing the CP version , the courses were saved automatically . 22.03.2022, 16:27, "Tensuko" ***@***.***>: Still don't know what you mean.No course is saved automatically.—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you commented.Message ID: ***@***.***>

**Reply by Tensuko** [reply]
No it won't.
Temporary courses are kept, yes, but that is not saved.

And my temporary courses are kept when I change arround on the Versions and I do that a lot.

**Comment by Serg12-22** [comment]
Thanks ! The main thing is that I found out where they are stored .

**Comment by wojtekczapeloo** [comment]
Hi, i have problem with harvesters (fendt katana from marketplace, and claas tucano 580 from modhub) close to trees it slows down to 2 km / h

i play MP, map maypole. problem is only on harvester @Tensuko

**Reply by Tensuko** [reply]
That is known and right now kind of working as intended.

**Comment by wojtekczapeloo** [comment]
But i Play on map where is a lot of tree and my harvester is very slow

śr., 23 mar 2022, 15:38 użytkownik Tensuko ***@***.***>
napisał:

**Reply by Tensuko** [reply]
Never understand why mapper place trees to close.
We won't change anything there for now.
Make your field smaler with field margin.

**Reply by rohne83** [reply]
> Never understand why mapper place trees to close. We won't change anything there for now. Make your field smaler with field margin.

Because it looks better and is more realistic? And I think there are very many players who prefer such realistic maps way over those artificial ones with vast empty grass areas around their fields. In reality you have dozens of trees directly at the edge of most fields.
Could you explain what the purpose of this slowdown is? To be honest, I also don't really understand that. The worker is following his course without significant side movement. So it will either hit the obstacle if it is inside the field boundary or it will pass it if it is outside... no matter at which speed.

**Reply by Tensuko** [reply]
> Because it looks better and is more realistic?

Is that a question or an Opinion ?



> In reality you have dozens of trees directly at the edge of most fields.

Where ? How close ?



> Could you explain what the purpose of this slowdown is?

Collision prevention. Mostly for turns, but since we don't want to difference turn and course.
Like in turns check only this on course check only that, we kept it like that for now.
And the slow down is how close the object is.



> The worker is following his course without significant side movement. So it will either hit the obstacle if it is inside the field boundary or it will pass it if it is outside... no matter at which speed.

Turns, vehicles parked wrong and so on.

**Comment by Serg12-22** [comment]
Good afternoon! Please tell me if a course correction will be added as it was in FS19 , otherwise in some places the equipment does not capture the territory that it should and therefore it has to be completed manually .

**Reply by Tensuko** [reply]
Not planed.

**Reply by Serg12-22** [reply]
Then tell me how to adjust the courses .

**Reply by Tensuko** [reply]
Change the generator settings.
There is not much need to adjust the course.

**Reply by Serg12-22** [reply]
Sometimes there are very crooked courses. And correction is also necessary when bypassing obstacles in the fields, when the equipment clings to them and stops. In FS19 , this helped a lot . About another thing: there is a stone collector with a width of 5 m. he does not want to go along the AutoDrive route after filling in, I have to manually turn on AutoDrive.

**Comment by Mc-Games** [comment]
Hello everyone, sometimes a course is not generated, the message "more in the Courseplay log file" where can I find this log file? Thanks for Support!
![CP_Logfile](https://user-images.githubusercontent.com/29283879/165704320-7b343bf0-c2f2-4a02-85d9-3078f96c3d51.png)

**Reply by Tensuko** [reply]
Thats the normal game logfile.
But I would just try to change parameters in course generator.

**Comment by Andy260785** [comment]
Hallo zusammen, 

ich habe ein Problem mit Courseplay. Es ladet mein Spielstand nur noch bis 85%.

In der Log steht folgender Fehler:
2022-05-18 23:02 Error: Running LUA method 'loadSharedI3DFileAsyncFinished'.
C:/Users/Andy7/OneDrive/Dokumente/My Games/FarmingSimulator2022/mods/FS22_Courseplay/scripts/specializations/CpInfoTexts.lua:49: attempt to index local 'spec' (a nil value)

Wäre toll wenn mir da jemand weiterhelfen könnte.

Vielen Dank schon mal dafür.

**Reply by XPiRiJenZ** [reply]
Hi,
versuch es mal hiermit:

    Spielstand OHNE CP starten
    Spielstand speichern
    Spielstand erneut MIT CP starten

Alternativ den letzten Mod, den du runter geladen hast, mal raus schmeißen. Wenn es vorher funktioniert hat, gibt es dann wohl damit ein Problem.

**Reply by Andy260785** [reply]
Hallo,

vielen Dank für die schnelle Antwort.

Mein Spielstand war zuvor ohne CP, und da hat alles Funktioniert, jetzt wo ich CP mit drauf habe funktioniert es nicht mehr.
Habe zwischenzeitlich CP mal wieder Deaktiviert, und da klappt es dann wieder.

Schöne Grüße

**Reply by Tensuko** [reply]
Hört sich für mich nach eine falschen CP Version oder sowas an.
Wo hast du die Version runter geladen, bzw hänge mal dein logfile an.
(Bitte NICHT den text einfach kopieren, drag and drop).

**Reply by Andy260785** [reply]
Hallo,

ich habe die aktuelle Version von der github Seite runtergeladen.
Bin gerade in der Arbeit, Hänge heute Abend den logfile mal an.
Kann es sein das das ganze Probleme macht, weil bei mir alles über OneDrive läuft?

**Reply by Tensuko** [reply]
Ist nicht auszuschließen.

**Comment by Andy260785** [comment]
Okay, dann muss ich mal schauen wie ich das bei Win10 deaktivieren kann.
Anbei noch mein logfile.
[log.txt](https://github.com/Courseplay/Courseplay_FS22/files/8739913/log.txt)

**Reply by Tensuko** [reply]
Du musst dein Spiel unbedingt updaten:
Version: 1.4.0.0 b16523
Wundert mich das du bei Steam noch immer die alter 1.4 hast, die 1.4.1 mit HotFix ist wichtig!

**Reply by Andy260785** [reply]
Okay, ist mir noch gar nicht aufgefallen. Aber stimmt das ist noch der alte Patch gewesen. Weis nicht warum es bei mir nicht aktualisiert wurde. habe das Spiel jetzt nochmal neu Installiert, und jetzt ist es der neue Patch. 
Spiel neu gestartet, und siehe da, jetzt funktioniert alles :-).
Hätte ich auch selber drauf kommen können, aber anscheinend den Wald vor lauter Bäume nicht gesehen ;-)
Jedenfalls vielen vielen Dank für euere Hilfe!

**Comment by farmer26** [comment]
I have looked and looked for a way for course play to load and unload wagons from a forage harvester similar to how a wagon will unload a combine.  I used forage pick up wagons with unload combine feature and it doesn't work.  May be I should try  a forage wagon without a pick up feature.  Can someone point me in the right direction?  Is this possible?

**Reply by Tensuko** [reply]
Combine only. No choppers.

**Reply by MishanyaKV** [reply]
Hello everyone who downloaded the latest version, did it change in your game? I still have 7.2.0.5. The files were replaced

**Reply by Tensuko** [reply]
And from where did you Download it? I am pretty sure you get it frim any shitty webside and not from our offizial release side.

**Reply by MishanyaKV** [reply]
I always download your mods from github

**Reply by Tensuko** [reply]
You need to load from release side.
Release is automatic created when we change to version so it have to be changes to correct version.

**Reply by farmer26** [reply]
> Hello everyone who downloaded the latest version, did it change in your game? I still have 7.2.0.5. The files were replaced

Check to make sure you don't have two CP zip files in your mod folder.  The new one may have been named ...CP..(1).zip

**Reply by MishanyaKV** [reply]
Sorry, this is my fault, I downloaded and installed the entire mods folder in documents, and the active folder with mods for the map was selected by dragging into the FSmods assistant program =)

**Reply by Tensuko** [reply]
Ok when it works now its good.

**Comment by DavidCriptoo** [comment]
infelizmente o sistema de quando acaba as sementes ou fertilizantes nas plantadeiras não funciona em conjunto com autodrive, os trabalhadores não vai reabastecer  a plantadeiras.

**Reply by Tensuko** [reply]
They do, ask AD on how it works.
